tests/phpunit/Differ/DiffColorizerTest.php

```diff
@@ -218,5 +218,37 @@ protected function name()
                 <diff-add>+        $changedLinesRangeXX = <diff-add-inline>2</diff-add-inline>;</diff-add></code>
                 CODE,
         ];
+
+        yield 'multibyte characters full-deletion' => [
+            <<<'CODE'
+                     function ($a) {
+                -        return 'déjà_vu';
+                +
+                     }
+                CODE,
+            <<<'CODE'
+                <code>
+                     function ($a) {
+                <diff-del>-<diff-del-inline>        return 'déjà_vu</diff-del-inline>';</diff-del>
+                <diff-add>+</diff-add>
+                     }</code>
+                CODE,
+        ];
+
+        yield 'multibyte characters partial-deletion' => [
+            <<<'CODE'
+                     function ($a) {
+                -        return 'déjà_vu' . 'bar';
+                +        return 'déjà_vu';
+                     }
+                CODE,
+            <<<'CODE'
+                <code>
+                     function ($a) {
+                <diff-del>-        return 'déjà_v<diff-del-inline>u' . 'ba</diff-del-inline>r';</diff-del>
```

Comment:
tbh I don't know whether this results are correct, but at least they reflect what the current implementation does

Reply:
Looks suspicious, agree. I think there is a bug in the algorithm and utf-8 characters, but since we are not adding/changing business logic here, let's merge and solve/debug it in the future